### PR TITLE
Classifier server

### PR DESCRIPTION
**Pull request by jpan17**
- run_classifier.py loads the classifier model and runs it on new data from app.js
- app_server.py manages the connection between run_classifier.py and app.js
- app.js sends new data to run_classifier.py and receives its predictions

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -94,22 +94,46 @@ function checkFinger(img, prediction, numFingers) {
...
+  // Pass these landmarks to the classifier
+  return fetch('http://127.0.0.1:8000', {
+    method: 'POST', 
+    body: JSON.stringify(landmarks)
+  }).then(
+      response => response.text()
+  ).then(
+      body => {
+          console.log(body)
+          correct = (parseInt(body) === numFingers)
```
Comment (WillPower264) [inline app.js:107]: this could just be `const correct = ...` and you can remove line 97

**Review by hollisma — COMMENTED** [review]
very noice

File: app.js [hunk 2]
```diff
@@ -94,22 +94,46 @@ function checkFinger(img, prediction, numFingers) {
...
-  return quadrantCorrect && checkFinger(img, prediction, target.numFingers);    
+    if (!prediction) {
+        return false
+    }
+    const quadrantCorrect = checkQuadrant(img, prediction.boundingBox, target.quadrant);
+    return checkFinger(img, prediction, target.numFingers).then(
+        correct => {
+            return correct && quadrantCorrect
+        }
+    );
```
Comment (WillPower264) [inline app.js:127]: we want to do `quadrantCorrect && something` so the api call doesn't happen if the quadrant is already wrong. i think this should be
```
return quadrantCorrect && checkFinger(img, prediction, target.numFingers)
```
which i now see is what i originally had, but i think you fixed the promise issue in `checkImage`.
can you change this and test if it still works?

**Review by WillPower264 — CHANGES_REQUESTED** [review]

File: app.js (hunk 2 above)
Comment (jpan17) [reply app.js:127]: Wooo it still works!

File: app.js [hunk 3]
```diff
@@ -120,7 +119,7 @@ function checkQuadrantAndFinger(img, prediction, target) {
         return false
     }
     const quadrantCorrect = checkQuadrant(img, prediction.boundingBox, target.quadrant);
-    return checkFinger(img, prediction, target.numFingers).then(
+    return quadrantCorrect && checkFinger(img, prediction, target.numFingers).then(
         correct => {
             return correct && quadrantCorrect
         }
```
Comment (WillPower264) [inline app.js]: no need to include `quadrantCorrect` again, and at that point it's just `correct => { return correct }`, which is just... nothing?

Comment (WillPower264) [reply app.js]: can you test if it works without the entire `.then(...)` part?

Comment (jpan17) [reply app.js]: Whoa you're right it didn't break

Comment (WillPower264) [reply app.js]: haha yayy, you fixed the problem in the other method, so this was fine to revert to what i originally had

**Review by WillPower264 — APPROVED** [review]

**Comment by WillPower264** [comment]
lgtm 👍

**Pull request merged**